query-engine/driver-adapters/executor/src/bench.ts

```diff
@@ -0,0 +1,204 @@
+/**
+ * Run with: `node --experimental-wasm-modules ./example.js`
+ * on Node.js 18+.
+ */
+import { webcrypto } from "node:crypto";
+import * as qe from "./qe";
+
+import pgDriver from "pg";
+import * as prismaPg from "@prisma/adapter-pg";
+import { bindAdapter, DriverAdapter } from "@prisma/driver-adapter-utils";
+
+import { recording } from "./recording";
+import prismaQueries from "../bench/queries.json";
+
+import { run, bench, group, baseline } from "mitata";
+
+import { QueryEngine as WasmBaseline } from "query-engine-wasm-baseline";
+import { QueryEngine as WasmLatest } from "query-engine-wasm-latest";
+
+(global as any).crypto = webcrypto;
+
+async function main(): Promise<void> {
+  // read the prisma schema from stdin
+  const datamodel = await new Promise<string>((resolve, reject) => {
+    let data = "";
+    process.stdin.on("data", (chunk) => {
+      data += chunk;
+    });
+    process.stdin.on("end", () => {
+      resolve(data);
+    });
+    process.stdin.on("error", reject);
+  });
+
+  const url = process.env.DATABASE_URL;
+  if (url == null) {
+    throw new Error("DATABASE_URL is not defined");
+  }
+  const pg = await pgAdapter(url);
+  const withErrorCapturing = bindAdapter(pg);
+
+  const { recorder, replayer } = recording(withErrorCapturing);
+  await recordQueries(recorder, datamodel, prismaQueries);
+  await benchMarkQueries(replayer, datamodel, prismaQueries);
+}
+
+async function recordQueries(
+  adapter: DriverAdapter,
+  datamodel: string,
+  prismaQueries: any
+): Promise<void> {
+  const qe = await initQeWasmBaseLine(adapter, datamodel);
+  await qe.connect("");
+
+  for (const prismaQuery of prismaQueries) {
+    const { description, query } = prismaQuery;
+    const res = await qe.query(JSON.stringify(query), "", undefined);
+
+    const errors = JSON.parse(res).errors;
+    if (errors != null && errors.length > 0) {
+      throw new Error(
+        `Query failed for ${description}: ${JSON.stringify(res)}`
+      );
+    }
+  }
+
+  await qe.disconnect("");
+}
+
+async function benchMarkQueries(
+  adapter: DriverAdapter,
+  datamodel: string,
+  prismaQueries: any
+) {
+  const napi = await initQeNapiCurrent(adapter, datamodel);
+  await napi.connect("");
+  const wasmCurrent = await initQeWasmCurrent(adapter, datamodel);
+  await wasmCurrent.connect("");
+  const wasmBaseline = await initQeWasmBaseLine(adapter, datamodel);
+  await wasmBaseline.connect("");
+  const wasmLatest = await initQeWasmLatest(adapter, datamodel);
+  await wasmLatest.connect("");
+
+  for (const prismaQuery of prismaQueries) {
+    const { description, query } = prismaQuery;
+
+    const adapters = {
```

Comment:
```suggestion
    const engines = {
```

and so on below. Naming confused me a little bit and I was struggling to understand why we need `JSON.parse` the result of the adapter until I realized it's actually the engine.

Reply:
Indeed, this was an oversight. Fixing too.